eng/pipelines/templates/jobs/sdk-build.yml

```diff
@@ -108,6 +108,7 @@ jobs:
         displayName: Build
         env:
           BuildConfig: $(buildConfiguration)
+          OPENSSL_ENABLE_SHA1_SIGNATURES: 1
```

Comment:
Should this be set on every non-Windows build? I don't know what this does so I don't know if it matters or not.

Reply:
From Michael:
_We originally set OPENSSL_ENABLE_SHA1_SIGNATURES=1 in our Fedora images._
_We no longer set this because we don't want it enabled for testing source-build.  Repos that need it should enable it within their environment_ See https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1476

So the recommendation was to set it in places we need it which our repo has some known and approved SHA1 usage. Only fedora41 docker containers are currently blocking it but I don't think there is harm in having it enabled for all linux since we need to be able to use SHA1 in our builds (we use it for WiX GUID creation in line with the UUID version 5 algorithm which is not used for security but rather setup upgrade consistency)